Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -0,0 +1,140 @@
+    public let schema: URL
+    /// Glob of files to search for GraphQL operations. This should be used to find queries and any client schema extensions. Defaults to `./**/*.graphql`, which will search for `.graphql` files throughout all subfolders of the folder where the script is run.
+    public let glob: String
+    /// [optional] Path to an operation id JSON map file. If specified, also stores the operation IDs (hashes) as properties on operation types. Defaults to `nil`.
+    public let operationIDs: URL?
+  }
+
+  /// Configure the folder structure of the generated files.
+  public struct FileOutput {
+    /// The location structure for the schema types files.
+    public let schemaTypes: SchemaTypesFileOutput
+    /// The location structure for the operation object files.
+    public let operations: OperationsFileOutput
+  }
+
+  /// Defines the location structure for the schema types files.
+  public struct SchemaTypesFileOutput {
+    /// Compatible dependency manager automation.
+    public enum DependencyManagerAutomation {
+      /// No automation. The generated schema types files must be manually added to the main application target.
+      case includedInTarget
+      /// Generate a podspec file.
+      case cocoaPods
+      /// Generate a cartfile.
+      case carthage
+      /// Generate a package.swift file.
+      case swiftPackageManager
+    }
+
+    /// The namespace to use for generated operation objects. When combined with `.includedInTarget` this should be the main target name. When used with any other dependency manager this should be the name of the new shared module that will be created.
+    public let name: String
+    /// An absolute location where the schema types files should be generated.
+    public let url: URL
+    /// Automation to ease the integration of the generated schem types file with compatible dependency managers.
+    public let dependencyAutomation: DependencyManagerAutomation
+  }
+
+  /// Defines the location structure for the operation object files.
+  public enum OperationsFileOutput {
+    /// All operation object files will be located in the module with the schema types.
+    case inSchemaModule
+    /// Operation object files will be co-located relative to the defining `.graphql` file.
+    case relative(subpath: String?)
+    /// All operation object files will be located in the specified path.
+    case absolute(url: URL)
+  }
+
+  /// Specify the formatting of the GraphQL query string literal.
+  public enum GraphQLQueryStringLiteralFormat {
+    /// The query string will be copied into the operation object with all line break formatting removed.
+    case singleLine
+    /// The query string will be copied with original formatting into the operation object.
+    case multiline
+    /// The query string will not be copied into the operation object. You must be using preregisterd queries with this configuration.
+    case exclude
+  }
+
+  public enum CompositionOption {
```

Comment:
I love this.  Great idea!

Reply:
can you add more docs on this particular one, because I'm not 100% sure what's happening here

Reply:
I intentionally didn't document the enum because I felt it was self-explanatory when looking at it in isolation. It takes on meaning in its application. I'll give it another pass though.

Reply:
I added it as a way to give more expression to true/false type properties. I generalized it to `CompositionOption` because it is used in multiple properties.

An example is `let omitDeprecatedEnumCases: Bool = true` for which you had to read/grok into the property name to fully understand what `true` and `false` meant. It now becomes `let deprecatedEnumCases: CompositionOption = .include` which IMO is a more expressive way to make the statement. The `Bool` probably makes evaluation of it more terse when you actually have to take action based on the value but I favour the readability of the enum.

I still can't think of a good description for the enum itself but I've added clarifying documentation to the properties of that type; see e426028f.

Reply:
Got it, this makes more sense now. I think it's throwing in a doc-comment on the type that this is to make what used to be boolean properties for whether to include or exclude certain things easier to reason about, because `CompositionOption` just as a name didn't really jump out to me.